### PR TITLE
Fix bugs in scrape.js and pre2017.js

### PR DESCRIPTION
**Pull request by li-boxuan**
scrape.js: Check if property repositories exists 
Fixes https://github.com/coala/gci-leaders/issues/160
pre2017.js: Fix wikipedia scraping
Fixes https://github.com/coala/gci-leaders/issues/162
scrape.js: Catch exception for GitHub GraphQL
Fixes https://github.com/coala/gci-leaders/issues/163

### CODE REVIEW

**Comment by li-boxuan** [comment]
Sometimes I encounter the issue  `(node:1599) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'line' of undefined`. See https://app.netlify.com/sites/gci-leaders/deploys/5bd2b2ddb13fb157f7a0a066

Probably an upstream bug with `graphql-client`. Retry sometimes solves the problem.

UPDATE:

So from my experiment and observation, it is a bug with upstream project `graphql-client`.

https://github.com/nordsimon/graphql-client/blob/f8d1dcba8a65f290e35b71c883a18094594d6d5a/index.js#L2

can lead to an error when elements in `errors` array don't have `locations` field.

However, GitHub is supposed to include the `locations` field when returning `errors`.

For example, a normal `errors` should be sth like this:

```javascript
[{"message":"Could not resolve to an Organization with the login of 'GCI-AOSSIE'.","type":"NOT_FOUND","path":["organization"],"locations":[{"line":2,"column":3}]}]
```

However, sometimes GitHub returns this:

```javascript
[{"message":"Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `E689:1AF6:67E816:BAAF4C:5BD321E1` when reporting this issue."}]
```

which causes the error of `graphql-client`.

CONCLUSION: a bug of upstream project `graphql-client`. It only happens occasionally, coz it's caused due to some unknown error of GitHub.

UPDATE 2:

Created an issue https://github.com/coala/gci-leaders/issues/163

File: tests/__data__/pre2017.json [hunk 1]
```diff
@@ -1,11 +1,7 @@
 [
   {
     "year": 2016,
-    "statistics": {
-      "number_of_organizations": 17,
-      "number_of_participants": 1340,
-      "total_tasks_completed": 6418
-    },
+    "statistics": {},
```
Comment (andrewda) [inline tests/__data__/pre2017.json]: This seems potentially unrelated?

Comment (li-boxuan) [reply tests/__data__/pre2017.json]: Yep. I just find out it is another bug. Created an issue for that https://github.com/coala/gci-leaders/issues/162

**Review by andrewda — COMMENTED** [review]
Carrying this over from @jayvdb's comment in #157, but it still says 2017 in the title.

Is this to be merged after #157?

**Comment by li-boxuan** [comment]
@andrewda Yep, this is to be merged after #157.

**Comment by jayvdb** [comment]
@li-boxuan , can you add a commit to fix the year. We gotta get this up . Can't wait any longer.

**Comment by li-boxuan** [comment]
@jayvdb Done. Without `moban` it is difficult to make the year in static files dynamic.

**Review by andrewda — APPROVED** [review]
👍 Good for now, we should make a task to implement dynamic years in page title/header.

**Comment by utkarsh2102** [comment]
Hey,
@li-boxuan, I've made it dynamic.
Please check: https://github.com/coala/gci-leaders/pull/157 
CC: @andrewda @jayvdb

**Comment by li-boxuan** [comment]
@utkarsh2102 Thanks. I've rebased this PR on top of yours.

**Comment by jayvdb** [comment]
ack https://github.com/coala/gci-leaders/pull/161/commits/e204e472f8d2ca15956d606bd937adad6750d044 https://github.com/coala/gci-leaders/pull/161/commits/ae47a1835b87988a81a21ab9ae682c219f93a7aa https://github.com/coala/gci-leaders/pull/161/commits/2ea74bc25661ac8f20028555df1d268a780f0c9f

**Review by jayvdb — APPROVED** [review]

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**